src/p2p/net_node.inl

```diff
@@ -826,6 +826,61 @@ void node_server<t_payload_net_handler>::update_peer_stats(const peerid_type pee
     stats.last_connected_timestamp = time(nullptr);
 }
 //-----------------------------------------------------------------------------------
+template<class t_payload_net_handler>
+uint64_t node_server<t_payload_net_handler>::calculate_peer_score(const peerid_type peer_id) {
+    constexpr uint64_t SCALE_FACTOR = 1000;
+    constexpr uint64_t NEUTRAL_SCORE = 100;
```

Comment:
Since none of this matters for consensus, I'd suggest changing this to score using `double`s so that you don't have to worry about scaling and integer truncation issues.

Reply:
Or overflow.  I think right now it's possible for the final score calculation to be negative (if `recent_activity_factor > 4*success_rate + 5*connection_duration_factor`) which would end up getting stuck with a node that actually should have a bad score.